### ISSUE TITLE
Panic on unwrap of None

### ISSUE THREAD

**Issue by tkaitchuck** [bheisler/criterion.rs#269]
When running the latest version from git, I get the following panic once all the benchmarks have run:
```
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', src/libcore/option.rs:355:21
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::panicking::default_hook::{{closure}}
             at src/libstd/sys_common/backtrace.rs:71
             at src/libstd/sys_common/backtrace.rs:59
             at src/libstd/panicking.rs:211
   2: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:227
             at src/libstd/panicking.rs:491
   3: std::panicking::continue_panic_fmt
             at src/libstd/panicking.rs:398
   4: rust_begin_unwind
             at src/libstd/panicking.rs:325
   5: core::panicking::panic_fmt
             at src/libcore/panicking.rs:95
   6: core::panicking::panic
             at src/libcore/panicking.rs:59
   7: <core::iter::Map<I, F> as core::iter::iterator::Iterator>::next
   8: <criterion::html::Html as criterion::report::Report>::final_summary
   9: <criterion::report::Reports as criterion::report::Report>::final_summary
```
This re-occurs each time, even if I do a `cargo clean`.

**Comment by bheisler** [comment]
Hey, thanks for the patience - I've been on vacation for a week and a half or so.

I haven't seen this one before. Could you share the benchmarks that are causing this issue?

**Comment by fitzgen** [comment]
I'm also hitting this. I'll see if I can make a minimal test case.

**Comment by mojzu** [comment]
I've run into this, here's the panic output I received.

```
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', src/libcore/option.rs:345:21
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:39
   1: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:71
   2: std::panicking::default_hook::{{closure}}
             at src/libstd/sys_common/backtrace.rs:59
             at src/libstd/panicking.rs:197
   3: std::panicking::default_hook
             at src/libstd/panicking.rs:211
   4: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:474
   5: std::panicking::continue_panic_fmt
             at src/libstd/panicking.rs:381
   6: rust_begin_unwind
             at src/libstd/panicking.rs:308
   7: core::panicking::panic_fmt
             at src/libcore/panicking.rs:85
   8: core::panicking::panic
             at src/libcore/panicking.rs:49
   9: <core::iter::adapters::Map<I, F> as core::iter::traits::iterator::Iterator>::fold
  10: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T, I>>::from_iter
  11: core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &mut F>::call_once
  12: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T, I>>::from_iter
  13: <criterion::html::Html as criterion::report::Report>::final_summary
  14: <criterion::report::Reports as criterion::report::Report>::final_summary
  15: criterion::Criterion::final_summary
  16: driver::main
  17: std::rt::lang_start::{{closure}}
  18: std::panicking::try::do_call
             at src/libstd/rt.rs:49
             at src/libstd/panicking.rs:293
  19: __rust_maybe_catch_panic
             at src/libpanic_unwind/lib.rs:87
  20: std::rt::lang_start_internal
             at src/libstd/panicking.rs:272
             at src/libstd/panic.rs:388
             at src/libstd/rt.rs:48
  21: main
  22: __libc_start_main
  23: _start
error: bench failed
```

In my case the cause was switching from a Benchmark to a ParameterizedBenchmark, after deleting the old data and rerunning the benchmarks the panic didn't occur.

**Comment by fitzgen** [comment]
https://github.com/fitzgen/criterion-issue-269

Running `cargo bench` in this repo will reproduce the unwrap of a `None` inside `final_summary` for me.

**Comment by fitzgen** [comment]
If I make that repo's parameterised benchmark be parameterised over only a single `n`, then it does not panic. If I remove the non-parameterised benchmark, then it does not panic.

Can't figure out how to get the test case any smaller than this.

**Comment by bheisler** [comment]
Thanks for the test case. Unfortunately due to some IRL stuff I probably won't get a chance to look at this in much detail for a while.

**Comment by mickvangelderen** [comment]
Probably ran into this issue as well. 

> In my case the cause was switching from a Benchmark to a ParameterizedBenchmark, after deleting the old data and rerunning the benchmarks the panic didn't occur

This was true and worked for me too.

**Comment by vmx** [comment]
I ran into the same issue and debugged a bit before I found the workaround here.

It happens here https://github.com/bheisler/criterion.rs/blob/cc4d5e841b4855db08d2501eee34f560d580880a/src/html/mod.rs#L520-L523

I changed the code to find where specifically it fails. It's the `collect()`:

```rust
        let groups1 = id_groups
            .into_iter();
        let groups2 = groups1
            .map(|(_, group)| BenchmarkGroup::new(output_directory, &group));
// This call fails
        let groups3 = groups2
            .collect::<Vec<BenchmarkGroup>>();
        let mut groups = groups3;
```

**Comment by tkaitchuck** [comment]
Running `cargo bench -- <target>` on github.com/tkaitchuck/ahash will result in this error.

**Comment by bheisler** [comment]
I think this ought to be fixed now; could you folks try it again with the latest git version? If you still see unwraps or other panics it's likely to be a different problem, so please post a new issue with the details.

Thanks for your patience, everyone.

**Comment by fitzgen** [comment]
Confirmed fixed with my test case -- thanks!

**Comment by fitzgen** [comment]
Would you mind cutting a new release with this bug fix when you get a chance?

**Comment by bheisler** [comment]
Published 0.2.11.

**Comment by fitzgen** [comment]
Thank you!
